sprocs/sync_assessments.sql

```diff
@@ -167,13 +167,30 @@ BEGIN
 
         -- if it is a group work try to insert a group_config
         IF (valid_assessment.data->>'group_work')::boolean THEN
-            INSERT INTO group_configs
-                (course_instance_id,
-                assessment_id
+            INSERT INTO group_configs (
+                course_instance_id,
+                assessment_id,
+                maximum,
+                minimum,
+                student_authz_create,
+                student_authz_join,
+                student_authz_leave
             ) VALUES (
                 syncing_course_instance_id,
-                new_assessment_id
-            ) ON CONFLICT DO NOTHING;
+                new_assessment_id,
+                (valid_assessment.data->>'group_max_size')::bigint,
+                (valid_assessment.data->>'group_min_size')::bigint,
+                (valid_assessment.data->>'student_group_create')::boolean,
+                (valid_assessment.data->>'student_group_join')::boolean,
+                (valid_assessment.data->>'student_group_leave')::boolean
+            ) ON CONFLICT (assessment_id) WHERE deleted_at IS NULL
```

Comment:
Do we need to be more careful with setting and clearing the `deleted_at` flag on the `group_configs` row? I think that if we UPDATE then we should set `deleted_at = NULL`, and if we don't have `group_work` then we should set `deleted_at = now()`?

Reply:
That is, I think we should remove the `WHERE deleted_at IS NULL` from the line above.

Reply:
For this part, I have to write "WHERE deleted_at IS NULL" here because our UNIQUE_INDEX is on `assessment_id WHERE deleted_at IS NULL`, which allows multiple deleted configs but only one unique current config. It won't let me complier if I don't put the statement here.

Reply:
Why do we have this constraint (`WHERE deleted_at IS NULL`) on the UNIQUE_INDEX? Could we drop this constraint and have the unique index be directly on `assessment_id`? This would allow us to delete and undelete group_configs as I suggested in the other comments.